internal/workflow.go

```diff
@@ -83,36 +99,69 @@ type (
 		//   decoding will be attempted, so you can try it yourself.
 		// - for other channels, an interface{} pointer.  All values are interfaces, so this will never fail, and you
 		//   can inspect the type with reflection or type assertions.
-		Receive(ctx Context, valuePtr interface{}) (more bool)
+		Receive(ctx Context, valuePtr interface{}) (ok bool)
 
-		// ReceiveAsync tries to receive from Channel without blocking.
+		// ReceiveAsync tries to Receive from Channel without blocking.
 		// If there is data available from the Channel, it assigns the data to valuePtr and returns true.
 		// Otherwise, it returns false immediately.
 		//
-		// This is equivalent to:
+		// This is technically equivalent to:
 		//  select {
-		//  case v := <- aChannel: ok = true
-		//  default: ok = false
+		//  case received, ok := <- aChannel:
+		//  	if ok {
+		//  		*valuePtr = received
+		//  	}
+		//  default:
+		//  	// no value was read
+		//  	ok = false
+		//  }
+		//
+		// But if your output values are zero values, this is equivalent to a simpler form:
+		//  select {
+		//  case value, ok := <- aChannel:
+		//  default:
+		//  	// no value was read
+		//  	ok = false
 		//  }
 		//
 		// Decoding or assigning failures are handled like Receive.
 		ReceiveAsync(valuePtr interface{}) (ok bool)
 
 		// ReceiveAsyncWithMoreFlag is the same as ReceiveAsync, with an extra return to indicate if there could be
-		// more value from the Channel. more is false when Channel is closed.
+		// more values from the Channel in the future.
+		// `more` is false only when Channel is closed and the read failed (empty).
 		//
-		// This is equivalent to:
+		// This is technically equivalent to:
 		//  select {
-		//  case v, more := <- aChannel: ok = true
-		//  default: ok = false
+		//  case received, ok := <- aChannel:
+		//  	if ok {
+		//  		*valuePtr = received
+		//  	}
+		//  	more = ok
+		//  default:
+		//  	// no value was read
+		//  	ok = false
+		//  	// but the read would have blocked, so the channel is not closed
+		//  	more = true
+		//  }
+		//
+		// But if your output values are zero values, this is equivalent to a simpler form:
+		//  select {
+		//  case value, ok := <- aChannel:
+		//  	more = ok
+		//  default:
+		//  	// no value was read
+		//  	ok = false
+		//  	// but the read would have blocked, so the channel is not closed
+		//  	more = true
 		//  }
```

Comment:
This took a bit of time to nail down precisely :|  it's a bit odd (I don't believe I've ever seen this kind of select statement in normal go code), but may be useful in some cases.